net/net.go

```diff
@@ -635,8 +634,6 @@ func (n *net) AddReplicator(
 		dialable, err = getDialable(paddr)
 		if err == nil {
 			n.host.Peerstore().AddAddr(pid, dialable, pstore.PermanentAddrTTL)
-		} else {
-			log.Warnf("peer %s address requires a DHT lookup", pid)
```

Comment:
This error can be misleading since the peer will only need a DHT lookup if the user supplied address doesn't include an IP/DNS portion _and_ there are no addresses in the peerstore. We're not checking the latter here.